### PR TITLE
Robustness: judge no launch.json exists.

### PR DESCRIPTION
**Pull request by Eskibear**
Pressing <kbd>Ctrl+F5</kbd> for a single file / first-time-opened project, the initial config to resolve is `{ noDebug: true }`. 
Current conditional doesn't apply. 

See more in  https://github.com/Microsoft/vscode-java-debug/issues/376#issuecomment-415677697

### CODE REVIEW

File: src/configurationProvider.ts [hunk 1]
```diff
@@ -101,11 +101,11 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
              * If no launch.json exists in the current workspace folder
              * delegate to provideDebugConfigurations api to generate the initial launch.json configurations
              */
-            if (Object.keys(config).length === 0 && folder !== undefined) {
+            if (!config.request && folder !== undefined) {
```
Comment (testforstephen) [inline src/configurationProvider.ts]: should write a common function to detect whether it's an empty config. Empty check should filter out the auto filled field `noDebug`.

File: src/configurationProvider.ts [hunk 2]
```diff
@@ -203,6 +203,11 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
         return !Array.isArray(configItems) || !configItems.length;
     }
 
+    private isEmptyConfig(config: vscode.DebugConfiguration): boolean {
```
Comment (yaohaizh) [inline src/configurationProvider.ts]: Should be isValidConfig

File: src/configurationProvider.ts [hunk 3]
```diff
@@ -203,6 +203,11 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
         return !Array.isArray(configItems) || !configItems.length;
     }
 
+    private isValidConfig(config: vscode.DebugConfiguration): boolean {
```
Comment (testforstephen) [inline src/configurationProvider.ts]: The intention is not for checking if config is valid or not, but find the case that launch.json is missing or empty. So isEmptyConfig is more appropriate.

Comment (Eskibear) [reply src/configurationProvider.ts]: > So isEmptyConfig is more appropriate.

After some consideration, I agree that `isValidConfig` might not be proper.  

Two reasons:
1. Confusion. `isValidConfig` implies the config contains all the information needed to launch a debug session, but actually it doesn't. 
2. For the name `isValidConfig`, it should have different value for following 2 configs, one is valid and the other invalid. However it doesn't, the function simply counts the keys and ignores "noDebug". 

```jsonc
// a valid config:
        {
            "type": "java",
            "name": "Debug (Launch)",
            "request": "launch", 
            "cwd": "${workspaceFolder}",
            "console": "internalConsole",
            "stopOnEntry": false,
            "mainClass": "",
            "args": ""
        }
// an invalid config:
       {
            "type": "java",
            "name": "Debug (Launch)",
            "request": "invalidValue",   // <------HERE
            "cwd": "${workspaceFolder}",
            "console": "internalConsole",
            "stopOnEntry": false,
            "mainClass": "",
            "args": ""
        }
```

For the name `isEmptyConfig`, yes we just regard `{noDebug: true}` as "empty", it's exactly why we need a helper function to customize the way we judge an empty config. Finally I'm ok with the name.
@yaohaizh are you ok with the name, or a better candidate if possible.

**Pull request merged**